{"cells":[{"cell_type":"code","execution_count":1,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":27161,"status":"ok","timestamp":1690695724318,"user":{"displayName":"박남규","userId":"16479626767028399096"},"user_tz":-540},"id":"r2tmeQxBUAwt","outputId":"33f126d6-2ccd-40a3-87ee-65bc692e9e6d"},"outputs":[{"name":"stdout","output_type":"stream","text":["Mounted at /content/drive\n"]}],"source":["from google.colab import drive\n","drive.mount('/content/drive')"]},{"cell_type":"code","execution_count":2,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":1112,"status":"ok","timestamp":1690695734803,"user":{"displayName":"박남규","userId":"16479626767028399096"},"user_tz":-540},"id":"ckPJ13DGUMK1","outputId":"043c7bd5-6eca-4b19-d470-0b04a2292b21"},"outputs":[{"name":"stdout","output_type":"stream","text":["/content/drive/My Drive/DL_project\n"]}],"source":["%cd /content/drive/My Drive/DL_project"]},{"cell_type":"code","execution_count":3,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":695,"status":"ok","timestamp":1690695744245,"user":{"displayName":"박남규","userId":"16479626767028399096"},"user_tz":-540},"id":"zj-MqrcwUYxM","outputId":"c46924d4-8174-417c-8b5f-295cf02aebd8"},"outputs":[{"name":"stdout","output_type":"stream","text":["\u001b[33mWARNING: Package(s) not found: ultralytics\u001b[0m\u001b[33m\n","\u001b[0m"]}],"source":["!pip show ultralytics"]},{"cell_type":"code","execution_count":4,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":4903,"status":"ok","timestamp":1690695763448,"user":{"displayName":"박남규","userId":"16479626767028399096"},"user_tz":-540},"id":"_aQZsSK3UbCZ","outputId":"f531a8b4-8f19-40f0-9996-a54cd3d0b5cf"},"outputs":[{"name":"stdout","output_type":"stream","text":["Collecting ultralytics\n","  Downloading ultralytics-8.0.145-py3-none-any.whl (605 kB)\n","\u001b[?25l     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m0.0/605.6 kB\u001b[0m \u001b[31m?\u001b[0m eta \u001b[36m-:--:--\u001b[0m\r\u001b[2K     \u001b[91m━━━━━━\u001b[0m\u001b[90m╺\u001b[0m\u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m92.2/605.6 kB\u001b[0m \u001b[31m3.0 MB/s\u001b[0m eta \u001b[36m0:00:01\u001b[0m\r\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m605.6/605.6 kB\u001b[0m \u001b[31m9.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hRequirement already satisfied: matplotlib\u003e=3.2.2 in /usr/local/lib/python3.10/dist-packages (from ultralytics) (3.7.1)\n","Requirement already satisfied: opencv-python\u003e=4.6.0 in /usr/local/lib/python3.10/dist-packages (from ultralytics) (4.7.0.72)\n","Requirement already satisfied: pillow\u003e=7.1.2 in /usr/local/lib/python3.10/dist-packages (from ultralytics) (9.4.0)\n","Requirement already satisfied: pyyaml\u003e=5.3.1 in /usr/local/lib/python3.10/dist-packages (from ultralytics) (6.0.1)\n","Requirement already satisfied: requests\u003e=2.23.0 in /usr/local/lib/python3.10/dist-packages (from ultralytics) (2.27.1)\n","Requirement already satisfied: scipy\u003e=1.4.1 in /usr/local/lib/python3.10/dist-packages (from ultralytics) (1.10.1)\n","Requirement already satisfied: torch\u003e=1.7.0 in /usr/local/lib/python3.10/dist-packages (from ultralytics) (2.0.1+cu118)\n","Requirement already satisfied: torchvision\u003e=0.8.1 in /usr/local/lib/python3.10/dist-packages (from ultralytics) (0.15.2+cu118)\n","Requirement already satisfied: tqdm\u003e=4.64.0 in /usr/local/lib/python3.10/dist-packages (from ultralytics) (4.65.0)\n","Requirement already satisfied: pandas\u003e=1.1.4 in /usr/local/lib/python3.10/dist-packages (from ultralytics) (1.5.3)\n","Requirement already satisfied: seaborn\u003e=0.11.0 in /usr/local/lib/python3.10/dist-packages (from ultralytics) (0.12.2)\n","Requirement already satisfied: psutil in /usr/local/lib/python3.10/dist-packages (from ultralytics) (5.9.5)\n","Requirement already satisfied: py-cpuinfo in /usr/local/lib/python3.10/dist-packages (from ultralytics) (9.0.0)\n","Requirement already satisfied: contourpy\u003e=1.0.1 in /usr/local/lib/python3.10/dist-packages (from matplotlib\u003e=3.2.2-\u003eultralytics) (1.1.0)\n","Requirement already satisfied: cycler\u003e=0.10 in /usr/local/lib/python3.10/dist-packages (from matplotlib\u003e=3.2.2-\u003eultralytics) (0.11.0)\n","Requirement already satisfied: fonttools\u003e=4.22.0 in /usr/local/lib/python3.10/dist-packages (from matplotlib\u003e=3.2.2-\u003eultralytics) (4.41.1)\n","Requirement already satisfied: kiwisolver\u003e=1.0.1 in /usr/local/lib/python3.10/dist-packages (from matplotlib\u003e=3.2.2-\u003eultralytics) (1.4.4)\n","Requirement already satisfied: numpy\u003e=1.20 in /usr/local/lib/python3.10/dist-packages (from matplotlib\u003e=3.2.2-\u003eultralytics) (1.22.4)\n","Requirement already satisfied: packaging\u003e=20.0 in /usr/local/lib/python3.10/dist-packages (from matplotlib\u003e=3.2.2-\u003eultralytics) (23.1)\n","Requirement already satisfied: pyparsing\u003e=2.3.1 in /usr/local/lib/python3.10/dist-packages (from matplotlib\u003e=3.2.2-\u003eultralytics) (3.1.0)\n","Requirement already satisfied: python-dateutil\u003e=2.7 in /usr/local/lib/python3.10/dist-packages (from matplotlib\u003e=3.2.2-\u003eultralytics) (2.8.2)\n","Requirement already satisfied: pytz\u003e=2020.1 in /usr/local/lib/python3.10/dist-packages (from pandas\u003e=1.1.4-\u003eultralytics) (2022.7.1)\n","Requirement already satisfied: urllib3\u003c1.27,\u003e=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests\u003e=2.23.0-\u003eultralytics) (1.26.16)\n","Requirement already satisfied: certifi\u003e=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests\u003e=2.23.0-\u003eultralytics) (2023.7.22)\n","Requirement already satisfied: charset-normalizer~=2.0.0 in /usr/local/lib/python3.10/dist-packages (from requests\u003e=2.23.0-\u003eultralytics) (2.0.12)\n","Requirement already satisfied: idna\u003c4,\u003e=2.5 in /usr/local/lib/python3.10/dist-packages (from requests\u003e=2.23.0-\u003eultralytics) (3.4)\n","Requirement already satisfied: filelock in /usr/local/lib/python3.10/dist-packages (from torch\u003e=1.7.0-\u003eultralytics) (3.12.2)\n","Requirement already satisfied: typing-extensions in /usr/local/lib/python3.10/dist-packages (from torch\u003e=1.7.0-\u003eultralytics) (4.7.1)\n","Requirement already satisfied: sympy in /usr/local/lib/python3.10/dist-packages (from torch\u003e=1.7.0-\u003eultralytics) (1.11.1)\n","Requirement already satisfied: networkx in /usr/local/lib/python3.10/dist-packages (from torch\u003e=1.7.0-\u003eultralytics) (3.1)\n","Requirement already satisfied: jinja2 in /usr/local/lib/python3.10/dist-packages (from torch\u003e=1.7.0-\u003eultralytics) (3.1.2)\n","Requirement already satisfied: triton==2.0.0 in /usr/local/lib/python3.10/dist-packages (from torch\u003e=1.7.0-\u003eultralytics) (2.0.0)\n","Requirement already satisfied: cmake in /usr/local/lib/python3.10/dist-packages (from triton==2.0.0-\u003etorch\u003e=1.7.0-\u003eultralytics) (3.25.2)\n","Requirement already satisfied: lit in /usr/local/lib/python3.10/dist-packages (from triton==2.0.0-\u003etorch\u003e=1.7.0-\u003eultralytics) (16.0.6)\n","Requirement already satisfied: six\u003e=1.5 in /usr/local/lib/python3.10/dist-packages (from python-dateutil\u003e=2.7-\u003ematplotlib\u003e=3.2.2-\u003eultralytics) (1.16.0)\n","Requirement already satisfied: MarkupSafe\u003e=2.0 in /usr/local/lib/python3.10/dist-packages (from jinja2-\u003etorch\u003e=1.7.0-\u003eultralytics) (2.1.3)\n","Requirement already satisfied: mpmath\u003e=0.19 in /usr/local/lib/python3.10/dist-packages (from sympy-\u003etorch\u003e=1.7.0-\u003eultralytics) (1.3.0)\n","Installing collected packages: ultralytics\n","Successfully installed ultralytics-8.0.145\n"]}],"source":["!pip install ultralytics"]},{"cell_type":"code","execution_count":5,"metadata":{"executionInfo":{"elapsed":997,"status":"ok","timestamp":1690695788409,"user":{"displayName":"박남규","userId":"16479626767028399096"},"user_tz":-540},"id":"3nJ2O8P9Uez_"},"outputs":[],"source":["from ultralytics import YOLO\n"]},{"cell_type":"code","execution_count":6,"metadata":{"executionInfo":{"elapsed":58735,"status":"ok","timestamp":1690695855376,"user":{"displayName":"박남규","userId":"16479626767028399096"},"user_tz":-540},"id":"hc3zYuKvUkQc"},"outputs":[],"source":["!unzip -qq '/content/drive/My Drive/DL_project/train2.zip'  -d '/content'"]},{"cell_type":"code","execution_count":7,"metadata":{"executionInfo":{"elapsed":16095,"status":"ok","timestamp":1690695886387,"user":{"displayName":"박남규","userId":"16479626767028399096"},"user_tz":-540},"id":"gVhYX8eOUoF7"},"outputs":[],"source":["!unzip -qq '/content/drive/My Drive/DL_project/valid2.zip'  -d '/content'"]},{"cell_type":"code","execution_count":8,"metadata":{"executionInfo":{"elapsed":2,"status":"ok","timestamp":1690695922708,"user":{"displayName":"박남규","userId":"16479626767028399096"},"user_tz":-540},"id":"jGqDw0pyU6Di"},"outputs":[],"source":["import torch\n","import os\n","import yaml\n","# 데이터셋 경로\n","train_data = '/content/train2'\n","val_data = '/content/valid2'\n","data_yaml = '/content/drive/My Drive/DL_project/data2.yaml'"]},{"cell_type":"code","execution_count":9,"metadata":{"executionInfo":{"elapsed":3,"status":"ok","timestamp":1690695927905,"user":{"displayName":"박남규","userId":"16479626767028399096"},"user_tz":-540},"id":"8amrGKtuVG3k"},"outputs":[],"source":["last_weight = '/content/drive/MyDrive/DL_project/runs/detect/train11/weights/last.pt'"]},{"cell_type":"code","execution_count":10,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":5950089,"status":"ok","timestamp":1690701888490,"user":{"displayName":"박남규","userId":"16479626767028399096"},"user_tz":-540},"id":"juZjJzE9VH6S","outputId":"0c81f6b5-b041-4e02-aad6-117a4699713f"},"outputs":[{"name":"stderr","output_type":"stream","text":["Ultralytics YOLOv8.0.145 🚀 Python-3.10.6 torch-2.0.1+cu118 CUDA:0 (Tesla T4, 15102MiB)\n","\u001b[34m\u001b[1mengine/trainer: \u001b[0mtask=detect, mode=train, model=/content/drive/MyDrive/DL_project/runs/detect/train11/weights/last.pt, data=/content/drive/My Drive/DL_project/data2.yaml, epochs=8, patience=50, batch=16, imgsz=416, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=None, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, show=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, vid_stride=1, line_width=None, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, boxes=True, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, cfg=None, tracker=botsort.yaml, save_dir=runs/detect/train12\n","Downloading https://ultralytics.com/assets/Arial.ttf to '/root/.config/Ultralytics/Arial.ttf'...\n","100%|██████████| 755k/755k [00:00\u003c00:00, 50.0MB/s]\n","\n","                   from  n    params  module                                       arguments                     \n","  0                  -1  1       464  ultralytics.nn.modules.conv.Conv             [3, 16, 3, 2]                 \n","  1                  -1  1      4672  ultralytics.nn.modules.conv.Conv             [16, 32, 3, 2]                \n","  2                  -1  1      7360  ultralytics.nn.modules.block.C2f             [32, 32, 1, True]             \n","  3                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]                \n","  4                  -1  2     49664  ultralytics.nn.modules.block.C2f             [64, 64, 2, True]             \n","  5                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               \n","  6                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]           \n","  7                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]              \n","  8                  -1  1    460288  ultralytics.nn.modules.block.C2f             [256, 256, 1, True]           \n","  9                  -1  1    164608  ultralytics.nn.modules.block.SPPF            [256, 256, 5]                 \n"," 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          \n"," 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           \n"," 12                  -1  1    148224  ultralytics.nn.modules.block.C2f             [384, 128, 1]                 \n"," 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          \n"," 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           \n"," 15                  -1  1     37248  ultralytics.nn.modules.block.C2f             [192, 64, 1]                  \n"," 16                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]                \n"," 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           \n"," 18                  -1  1    123648  ultralytics.nn.modules.block.C2f             [192, 128, 1]                 \n"," 19                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]              \n"," 20             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           \n"," 21                  -1  1    493056  ultralytics.nn.modules.block.C2f             [384, 256, 1]                 \n"," 22        [15, 18, 21]  1    756187  ultralytics.nn.modules.head.Detect           [25, [64, 128, 256]]          \n","Model summary: 225 layers, 3015723 parameters, 3015707 gradients\n","\n","Transferred 355/355 items from pretrained weights\n","\u001b[34m\u001b[1mTensorBoard: \u001b[0mStart with 'tensorboard --logdir runs/detect/train12', view at http://localhost:6006/\n","\u001b[34m\u001b[1mAMP: \u001b[0mrunning Automatic Mixed Precision (AMP) checks with YOLOv8n...\n","\u001b[34m\u001b[1mAMP: \u001b[0mchecks passed ✅\n","\u001b[34m\u001b[1mtrain: \u001b[0mScanning /content/train2... 63319 images, 0 backgrounds, 9 corrupt: 100%|██████████| 63319/63319 [01:20\u003c00:00, 784.36it/s] \n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/1032642.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0076]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/1032646.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0038]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/1032650.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [      1.001]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/123915.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0087]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/123968.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0013]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/123970.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0025]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/233925.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0025]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/68272.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [      1.005]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/70355.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0013]\n","\u001b[34m\u001b[1mtrain: \u001b[0mNew cache created: /content/train2.cache\n","\u001b[34m\u001b[1malbumentations: \u001b[0mBlur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01), CLAHE(p=0.01, clip_limit=(1, 4.0), tile_grid_size=(8, 8))\n","\u001b[34m\u001b[1mval: \u001b[0mScanning /content/valid2... 15830 images, 0 backgrounds, 5 corrupt: 100%|██████████| 15830/15830 [00:09\u003c00:00, 1621.68it/s]\n","\u001b[34m\u001b[1mval: \u001b[0mWARNING ⚠️ /content/valid2/1032641.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0067]\n","\u001b[34m\u001b[1mval: \u001b[0mWARNING ⚠️ /content/valid2/1032884.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0038]\n","\u001b[34m\u001b[1mval: \u001b[0mWARNING ⚠️ /content/valid2/123918.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0125]\n","\u001b[34m\u001b[1mval: \u001b[0mWARNING ⚠️ /content/valid2/233996.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0087]\n","\u001b[34m\u001b[1mval: \u001b[0mWARNING ⚠️ /content/valid2/54480.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0013]\n","\u001b[34m\u001b[1mval: \u001b[0mNew cache created: /content/valid2.cache\n","Plotting labels to runs/detect/train12/labels.jpg... \n","\u001b[34m\u001b[1moptimizer:\u001b[0m AdamW(lr=0.000345, momentum=0.9) with parameter groups 57 weight(decay=0.0), 64 weight(decay=0.0005), 63 bias(decay=0.0)\n","Image sizes 416 train, 416 val\n","Using 4 dataloader workers\n","Logging results to \u001b[1mruns/detect/train12\u001b[0m\n","Starting training for 8 epochs...\n","\n","      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size\n","        1/8      1.33G     0.7207      1.016      1.018         50        416: 100%|██████████| 3957/3957 [09:34\u003c00:00,  6.89it/s]\n","                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 495/495 [02:28\u003c00:00,  3.33it/s]\n","                   all      15825      25358      0.737        0.7      0.769      0.645\n","\n","      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size\n","        2/8      1.35G     0.6987     0.9723      1.008         46        416: 100%|██████████| 3957/3957 [09:18\u003c00:00,  7.08it/s]\n","                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 495/495 [02:24\u003c00:00,  3.43it/s]\n","                   all      15825      25358      0.722      0.684      0.751      0.628\n","\n","      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size\n","        3/8      1.35G     0.6989     0.9631      1.007         48        416: 100%|██████████| 3957/3957 [09:18\u003c00:00,  7.08it/s]\n","                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 495/495 [02:26\u003c00:00,  3.38it/s]\n","                   all      15825      25358      0.743      0.674      0.752      0.627\n","\n","      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size\n","        4/8      1.35G     0.7013     0.9785      1.008         43        416: 100%|██████████| 3957/3957 [09:20\u003c00:00,  7.07it/s]\n","                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 495/495 [02:27\u003c00:00,  3.35it/s]\n","                   all      15825      25358      0.736      0.682      0.755      0.633\n","\n","      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size\n","        5/8      1.36G     0.7015     0.9665      1.009         49        416: 100%|██████████| 3957/3957 [09:14\u003c00:00,  7.13it/s]\n","                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 495/495 [02:27\u003c00:00,  3.35it/s]\n","                   all      15825      25358      0.753      0.687      0.766      0.641\n","\n","      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size\n","        6/8      1.36G     0.7051     0.9813      1.011         61        416: 100%|██████████| 3957/3957 [09:16\u003c00:00,  7.12it/s]\n","                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 495/495 [02:27\u003c00:00,  3.36it/s]\n","                   all      15825      25358      0.765      0.695      0.776      0.651\n","\n","      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size\n","        7/8      1.37G     0.7116      0.993      1.014         46        416: 100%|██████████| 3957/3957 [09:10\u003c00:00,  7.18it/s]\n","                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 495/495 [02:26\u003c00:00,  3.39it/s]\n","                   all      15825      25358      0.773      0.701      0.786      0.662\n","\n","      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size\n","        8/8      1.35G     0.7244       1.02       1.02         47        416: 100%|██████████| 3957/3957 [09:11\u003c00:00,  7.17it/s]\n","                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 495/495 [02:37\u003c00:00,  3.15it/s]\n","                   all      15825      25358      0.782      0.703      0.797      0.672\n","\n","8 epochs completed in 1.573 hours.\n","Optimizer stripped from runs/detect/train12/weights/last.pt, 6.2MB\n","Optimizer stripped from runs/detect/train12/weights/best.pt, 6.2MB\n","\n","Validating runs/detect/train12/weights/best.pt...\n","Ultralytics YOLOv8.0.145 🚀 Python-3.10.6 torch-2.0.1+cu118 CUDA:0 (Tesla T4, 15102MiB)\n","Model summary (fused): 168 layers, 3010523 parameters, 0 gradients\n","                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 495/495 [02:31\u003c00:00,  3.26it/s]\n","                   all      15825      25358      0.782      0.703      0.797      0.672\n","               padding      15825        959      0.881      0.909      0.949      0.832\n","                   fur      15825        706      0.736      0.625      0.753      0.649\n","               Mustang      15825        221      0.775      0.624      0.788      0.684\n","                 suede      15825        446      0.686      0.567      0.654      0.535\n","              corduroy      15825        991      0.835      0.815      0.882      0.742\n","        Sequin/Glitter      15825         48      0.763      0.469      0.666      0.547\n","                 Denim      15825       2391      0.885      0.845      0.917      0.737\n","                jersey      15825       2148      0.703      0.605      0.717      0.584\n","                 tweed      15825       1021      0.878      0.899      0.946      0.796\n","                velvet      15825        649      0.802      0.786      0.867      0.733\n","             vinyl/PVC      15825        148      0.817      0.602      0.736      0.632\n","         wool/cashmere      15825       1177      0.744       0.67      0.771      0.649\n","             hair knit      15825        536      0.817      0.648      0.765       0.67\n","                  knit      15825       2186      0.744      0.597      0.726      0.594\n","                  lace      15825       1030      0.884      0.817      0.909      0.773\n","                 linen      15825       1028      0.722      0.689      0.772      0.655\n","                 messi      15825        328      0.771      0.597      0.715      0.579\n","                fleece      15825       1074      0.773      0.794      0.831      0.729\n","              neoprene      15825         93      0.646      0.669      0.706       0.63\n","                  silk      15825       1014      0.779       0.78       0.84      0.738\n","               spandex      15825        981      0.813      0.796      0.842      0.633\n","              jacquard      15825        258       0.74      0.554       0.68      0.583\n","               leather      15825        748      0.819      0.801      0.877       0.72\n","               chiffon      15825       1131      0.819      0.763      0.855      0.737\n","                 woven      15825       4046      0.725      0.658      0.761      0.628\n","Speed: 0.1ms preprocess, 1.2ms inference, 0.0ms loss, 0.9ms postprocess per image\n","Results saved to \u001b[1mruns/detect/train12\u001b[0m\n"]}],"source":["model = YOLO(last_weight)\n","\n","model.resume = True\n","\n","results = model.train(data =data_yaml, epochs=8, imgsz=416)"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"wJYknOsy96Pv"},"outputs":[],"source":[]},{"cell_type":"code","execution_count":11,"metadata":{"executionInfo":{"elapsed":2,"status":"ok","timestamp":1690702377638,"user":{"displayName":"박남규","userId":"16479626767028399096"},"user_tz":-540},"id":"sv3122rItrvF"},"outputs":[],"source":["last_weight = '/content/drive/MyDrive/DL_project/runs/detect/train12/weights/last.pt'"]},{"cell_type":"code","execution_count":12,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":5889351,"status":"ok","timestamp":1690708335966,"user":{"displayName":"박남규","userId":"16479626767028399096"},"user_tz":-540},"id":"HZsQE2T0VKth","outputId":"876fdad4-9937-4bb6-e7fd-e609dd7bebdc"},"outputs":[{"name":"stderr","output_type":"stream","text":["Ultralytics YOLOv8.0.145 🚀 Python-3.10.6 torch-2.0.1+cu118 CUDA:0 (Tesla T4, 15102MiB)\n","\u001b[34m\u001b[1mengine/trainer: \u001b[0mtask=detect, mode=train, model=/content/drive/MyDrive/DL_project/runs/detect/train12/weights/last.pt, data=/content/drive/My Drive/DL_project/data2.yaml, epochs=8, patience=50, batch=16, imgsz=416, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=None, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, show=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, vid_stride=1, line_width=None, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, boxes=True, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, cfg=None, tracker=botsort.yaml, save_dir=runs/detect/train13\n","\n","                   from  n    params  module                                       arguments                     \n","  0                  -1  1       464  ultralytics.nn.modules.conv.Conv             [3, 16, 3, 2]                 \n","  1                  -1  1      4672  ultralytics.nn.modules.conv.Conv             [16, 32, 3, 2]                \n","  2                  -1  1      7360  ultralytics.nn.modules.block.C2f             [32, 32, 1, True]             \n","  3                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]                \n","  4                  -1  2     49664  ultralytics.nn.modules.block.C2f             [64, 64, 2, True]             \n","  5                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               \n","  6                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]           \n","  7                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]              \n","  8                  -1  1    460288  ultralytics.nn.modules.block.C2f             [256, 256, 1, True]           \n","  9                  -1  1    164608  ultralytics.nn.modules.block.SPPF            [256, 256, 5]                 \n"," 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          \n"," 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           \n"," 12                  -1  1    148224  ultralytics.nn.modules.block.C2f             [384, 128, 1]                 \n"," 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          \n"," 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           \n"," 15                  -1  1     37248  ultralytics.nn.modules.block.C2f             [192, 64, 1]                  \n"," 16                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]                \n"," 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           \n"," 18                  -1  1    123648  ultralytics.nn.modules.block.C2f             [192, 128, 1]                 \n"," 19                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]              \n"," 20             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           \n"," 21                  -1  1    493056  ultralytics.nn.modules.block.C2f             [384, 256, 1]                 \n"," 22        [15, 18, 21]  1    756187  ultralytics.nn.modules.head.Detect           [25, [64, 128, 256]]          \n","Model summary: 225 layers, 3015723 parameters, 3015707 gradients\n","\n","Transferred 355/355 items from pretrained weights\n","\u001b[34m\u001b[1mTensorBoard: \u001b[0mStart with 'tensorboard --logdir runs/detect/train13', view at http://localhost:6006/\n","\u001b[34m\u001b[1mAMP: \u001b[0mrunning Automatic Mixed Precision (AMP) checks with YOLOv8n...\n","\u001b[34m\u001b[1mAMP: \u001b[0mchecks passed ✅\n","\u001b[34m\u001b[1mtrain: \u001b[0mScanning /content/train2.cache... 63319 images, 0 backgrounds, 9 corrupt: 100%|██████████| 63319/63319 [00:00\u003c?, ?it/s]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/1032642.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0076]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/1032646.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0038]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/1032650.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [      1.001]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/123915.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0087]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/123968.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0013]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/123970.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0025]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/233925.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0025]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/68272.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [      1.005]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/70355.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0013]\n","\u001b[34m\u001b[1malbumentations: \u001b[0mBlur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01), CLAHE(p=0.01, clip_limit=(1, 4.0), tile_grid_size=(8, 8))\n","\u001b[34m\u001b[1mval: \u001b[0mScanning /content/valid2.cache... 15830 images, 0 backgrounds, 5 corrupt: 100%|██████████| 15830/15830 [00:00\u003c?, ?it/s]\n","\u001b[34m\u001b[1mval: \u001b[0mWARNING ⚠️ /content/valid2/1032641.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0067]\n","\u001b[34m\u001b[1mval: \u001b[0mWARNING ⚠️ /content/valid2/1032884.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0038]\n","\u001b[34m\u001b[1mval: \u001b[0mWARNING ⚠️ /content/valid2/123918.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0125]\n","\u001b[34m\u001b[1mval: \u001b[0mWARNING ⚠️ /content/valid2/233996.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0087]\n","\u001b[34m\u001b[1mval: \u001b[0mWARNING ⚠️ /content/valid2/54480.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0013]\n","Plotting labels to runs/detect/train13/labels.jpg... \n","\u001b[34m\u001b[1moptimizer:\u001b[0m AdamW(lr=0.000345, momentum=0.9) with parameter groups 57 weight(decay=0.0), 64 weight(decay=0.0005), 63 bias(decay=0.0)\n","Image sizes 416 train, 416 val\n","Using 4 dataloader workers\n","Logging results to \u001b[1mruns/detect/train13\u001b[0m\n","Starting training for 8 epochs...\n","\n","      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size\n","        1/8      1.35G     0.7156      0.983      1.015         50        416: 100%|██████████| 3957/3957 [09:41\u003c00:00,  6.80it/s]\n","                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 495/495 [02:25\u003c00:00,  3.41it/s]\n","                   all      15825      25358      0.759      0.709      0.781      0.656\n","\n","      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size\n","        2/8      1.34G     0.6903     0.9258      1.004         46        416: 100%|██████████| 3957/3957 [09:21\u003c00:00,  7.04it/s]\n","                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 495/495 [02:26\u003c00:00,  3.38it/s]\n","                   all      15825      25358      0.735      0.692      0.756      0.633\n","\n","      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size\n","        3/8      1.35G     0.6894     0.9164      1.003         48        416: 100%|██████████| 3957/3957 [09:19\u003c00:00,  7.07it/s]\n","                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 495/495 [02:28\u003c00:00,  3.32it/s]\n","                   all      15825      25358      0.748      0.684      0.756      0.632\n","\n","      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size\n","        4/8      1.35G     0.6927     0.9337      1.004         43        416: 100%|██████████| 3957/3957 [09:23\u003c00:00,  7.02it/s]\n","                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 495/495 [02:24\u003c00:00,  3.42it/s]\n","                   all      15825      25358      0.744      0.692      0.764      0.641\n","\n","      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size\n","        5/8      1.35G     0.6929     0.9242      1.005         49        416: 100%|██████████| 3957/3957 [09:16\u003c00:00,  7.12it/s]\n","                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 495/495 [02:24\u003c00:00,  3.44it/s]\n","                   all      15825      25358      0.775      0.684      0.773      0.648\n","\n","      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size\n","        6/8      1.35G     0.6989     0.9429      1.007         61        416: 100%|██████████| 3957/3957 [09:16\u003c00:00,  7.11it/s]\n","                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 495/495 [02:25\u003c00:00,  3.40it/s]\n","                   all      15825      25358      0.756      0.707      0.779      0.655\n","\n","      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size\n","        7/8      1.35G     0.7059     0.9599      1.011         46        416: 100%|██████████| 3957/3957 [09:20\u003c00:00,  7.06it/s]\n","                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 495/495 [02:23\u003c00:00,  3.45it/s]\n","                   all      15825      25358      0.772      0.713      0.791      0.667\n","\n","      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size\n","        8/8      1.35G     0.7211     0.9962      1.019         47        416: 100%|██████████| 3957/3957 [09:13\u003c00:00,  7.15it/s]\n","                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 495/495 [02:33\u003c00:00,  3.22it/s]\n","                   all      15825      25358      0.779      0.724      0.802      0.677\n","\n","8 epochs completed in 1.577 hours.\n","Optimizer stripped from runs/detect/train13/weights/last.pt, 6.2MB\n","Optimizer stripped from runs/detect/train13/weights/best.pt, 6.2MB\n","\n","Validating runs/detect/train13/weights/best.pt...\n","Ultralytics YOLOv8.0.145 🚀 Python-3.10.6 torch-2.0.1+cu118 CUDA:0 (Tesla T4, 15102MiB)\n","Model summary (fused): 168 layers, 3010523 parameters, 0 gradients\n","                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 495/495 [02:28\u003c00:00,  3.33it/s]\n","                   all      15825      25358       0.78      0.722      0.802      0.677\n","               padding      15825        959      0.883      0.906      0.951      0.834\n","                   fur      15825        706      0.729      0.652      0.762       0.66\n","               Mustang      15825        221       0.78      0.652        0.8      0.695\n","                 suede      15825        446      0.693      0.576      0.659      0.539\n","              corduroy      15825        991      0.833      0.816      0.888       0.75\n","        Sequin/Glitter      15825         48      0.783      0.603      0.687      0.572\n","                 Denim      15825       2391      0.881      0.847      0.917      0.737\n","                jersey      15825       2148      0.691      0.632       0.72      0.587\n","                 tweed      15825       1021      0.868      0.893      0.946      0.798\n","                velvet      15825        649      0.817      0.789      0.866      0.734\n","             vinyl/PVC      15825        148      0.813      0.649      0.724      0.615\n","         wool/cashmere      15825       1177       0.73      0.689      0.779      0.658\n","             hair knit      15825        536        0.8      0.662      0.769      0.676\n","                  knit      15825       2186      0.729      0.611      0.732      0.599\n","                  lace      15825       1030      0.882      0.819      0.908      0.774\n","                 linen      15825       1028      0.729       0.71      0.782       0.66\n","                 messi      15825        328      0.753      0.622      0.714      0.578\n","                fleece      15825       1074       0.77      0.808      0.837      0.733\n","              neoprene      15825         93      0.667      0.677      0.738      0.659\n","                  silk      15825       1014      0.769      0.783      0.841       0.74\n","               spandex      15825        981      0.811      0.803      0.845       0.64\n","              jacquard      15825        258      0.746       0.59      0.697      0.599\n","               leather      15825        748       0.81       0.81      0.877      0.721\n","               chiffon      15825       1131      0.808      0.775      0.855       0.74\n","                 woven      15825       4046      0.721      0.668      0.764      0.631\n","Speed: 0.1ms preprocess, 1.2ms inference, 0.0ms loss, 0.9ms postprocess per image\n","Results saved to \u001b[1mruns/detect/train13\u001b[0m\n"]}],"source":["model = YOLO(last_weight)\n","\n","model.resume = True\n","\n","results = model.train(data =data_yaml, epochs=8, imgsz=416)"]},{"cell_type":"code","execution_count":13,"metadata":{"executionInfo":{"elapsed":2,"status":"ok","timestamp":1690709270901,"user":{"displayName":"박남규","userId":"16479626767028399096"},"user_tz":-540},"id":"J6gZKytEt_eA"},"outputs":[],"source":["last_weight = '/content/drive/MyDrive/DL_project/runs/detect/train13/weights/last.pt'"]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"background_save":true,"base_uri":"https://localhost:8080/"},"id":"azAS_BhoIBu0"},"outputs":[{"name":"stderr","output_type":"stream","text":["Ultralytics YOLOv8.0.145 🚀 Python-3.10.6 torch-2.0.1+cu118 CUDA:0 (Tesla T4, 15102MiB)\n","\u001b[34m\u001b[1mengine/trainer: \u001b[0mtask=detect, mode=train, model=/content/drive/MyDrive/DL_project/runs/detect/train13/weights/last.pt, data=/content/drive/My Drive/DL_project/data2.yaml, epochs=8, patience=50, batch=16, imgsz=416, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=None, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, show=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, vid_stride=1, line_width=None, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, boxes=True, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, cfg=None, tracker=botsort.yaml, save_dir=runs/detect/train15\n","\n","                   from  n    params  module                                       arguments                     \n","  0                  -1  1       464  ultralytics.nn.modules.conv.Conv             [3, 16, 3, 2]                 \n","  1                  -1  1      4672  ultralytics.nn.modules.conv.Conv             [16, 32, 3, 2]                \n","  2                  -1  1      7360  ultralytics.nn.modules.block.C2f             [32, 32, 1, True]             \n","  3                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]                \n","  4                  -1  2     49664  ultralytics.nn.modules.block.C2f             [64, 64, 2, True]             \n","  5                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               \n","  6                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]           \n","  7                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]              \n","  8                  -1  1    460288  ultralytics.nn.modules.block.C2f             [256, 256, 1, True]           \n","  9                  -1  1    164608  ultralytics.nn.modules.block.SPPF            [256, 256, 5]                 \n"," 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          \n"," 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           \n"," 12                  -1  1    148224  ultralytics.nn.modules.block.C2f             [384, 128, 1]                 \n"," 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          \n"," 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           \n"," 15                  -1  1     37248  ultralytics.nn.modules.block.C2f             [192, 64, 1]                  \n"," 16                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]                \n"," 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           \n"," 18                  -1  1    123648  ultralytics.nn.modules.block.C2f             [192, 128, 1]                 \n"," 19                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]              \n"," 20             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           \n"," 21                  -1  1    493056  ultralytics.nn.modules.block.C2f             [384, 256, 1]                 \n"," 22        [15, 18, 21]  1    756187  ultralytics.nn.modules.head.Detect           [25, [64, 128, 256]]          \n","Model summary: 225 layers, 3015723 parameters, 3015707 gradients\n","\n","Transferred 355/355 items from pretrained weights\n","\u001b[34m\u001b[1mTensorBoard: \u001b[0mStart with 'tensorboard --logdir runs/detect/train15', view at http://localhost:6006/\n","\u001b[34m\u001b[1mAMP: \u001b[0mrunning Automatic Mixed Precision (AMP) checks with YOLOv8n...\n","\u001b[34m\u001b[1mAMP: \u001b[0mchecks passed ✅\n","\u001b[34m\u001b[1mtrain: \u001b[0mScanning /content/train2.cache... 63319 images, 0 backgrounds, 9 corrupt: 100%|██████████| 63319/63319 [00:00\u003c?, ?it/s]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/1032642.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0076]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/1032646.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0038]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/1032650.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [      1.001]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/123915.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0087]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/123968.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0013]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/123970.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0025]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/233925.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0025]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/68272.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [      1.005]\n","\u001b[34m\u001b[1mtrain: \u001b[0mWARNING ⚠️ /content/train2/70355.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0013]\n","\u001b[34m\u001b[1malbumentations: \u001b[0mBlur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01), CLAHE(p=0.01, clip_limit=(1, 4.0), tile_grid_size=(8, 8))\n","\u001b[34m\u001b[1mval: \u001b[0mScanning /content/valid2.cache... 15830 images, 0 backgrounds, 5 corrupt: 100%|██████████| 15830/15830 [00:00\u003c?, ?it/s]\n","\u001b[34m\u001b[1mval: \u001b[0mWARNING ⚠️ /content/valid2/1032641.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0067]\n","\u001b[34m\u001b[1mval: \u001b[0mWARNING ⚠️ /content/valid2/1032884.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0038]\n","\u001b[34m\u001b[1mval: \u001b[0mWARNING ⚠️ /content/valid2/123918.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0125]\n","\u001b[34m\u001b[1mval: \u001b[0mWARNING ⚠️ /content/valid2/233996.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0087]\n","\u001b[34m\u001b[1mval: \u001b[0mWARNING ⚠️ /content/valid2/54480.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0013]\n","Plotting labels to runs/detect/train15/labels.jpg... \n","\u001b[34m\u001b[1moptimizer:\u001b[0m AdamW(lr=0.000345, momentum=0.9) with parameter groups 57 weight(decay=0.0), 64 weight(decay=0.0005), 63 bias(decay=0.0)\n","Image sizes 416 train, 416 val\n","Using 4 dataloader workers\n","Logging results to \u001b[1mruns/detect/train15\u001b[0m\n","Starting training for 8 epochs...\n","\n","      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size\n","        1/8      1.97G     0.7117     0.9571      1.013         50        416: 100%|██████████| 3957/3957 [09:34\u003c00:00,  6.89it/s]\n","                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 495/495 [02:27\u003c00:00,  3.36it/s]\n","                   all      15825      25358      0.787      0.695      0.786      0.661\n","\n","      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size\n","        2/8      1.85G     0.6922     0.9104      1.005         58        416:   8%|▊         | 312/3957 [00:45\u003c08:05,  7.50it/s]Exception in thread Thread-72:\n","Traceback (most recent call last):\n","  File \"/usr/lib/python3.10/threading.py\", line 1016, in _bootstrap_inner\n","    self.run()\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorboard/summary/writer/event_file_writer.py\", line 244, in run\n","    self._run()\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorboard/summary/writer/event_file_writer.py\", line 275, in _run\n","    self._record_writer.write(data)\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorboard/summary/writer/record_writer.py\", line 40, in write\n","    self._writer.write(header + header_crc + data + footer_crc)\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorflow/python/lib/io/file_io.py\", line 101, in write\n","    self._writable_file.append(\n","tensorflow.python.framework.errors_impl.FailedPreconditionError: runs/detect/train15/events.out.tfevents.1690709478.098cdce72173.289.3; Transport endpoint is not connected\n","        2/8      1.85G     0.6919     0.9098      1.005         47        416:   8%|▊         | 313/3957 [00:45\u003c08:47,  6.91it/s]\n","ERROR:root:Internal Python error in the inspect module.\n","Below is the traceback from this internal error.\n","\n","ERROR:root:Internal Python error in the inspect module.\n","Below is the traceback from this internal error.\n","\n","ERROR:root:Internal Python error in the inspect module.\n","Below is the traceback from this internal error.\n","\n"]},{"name":"stdout","output_type":"stream","text":["Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 3553, in run_code\n","    exec(code_obj, self.user_global_ns, self.user_ns)\n","  File \"\u003cipython-input-15-0c940d6c2741\u003e\", line 5, in \u003ccell line: 5\u003e\n","    results = model.train(data =data_yaml, epochs=8, imgsz=416)\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/engine/model.py\", line 377, in train\n","    self.trainer.train()\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/engine/trainer.py\", line 192, in train\n","    self._do_train(world_size)\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/engine/trainer.py\", line 354, in _do_train\n","    self.run_callbacks('on_batch_end')\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/engine/trainer.py\", line 165, in run_callbacks\n","    callback(self)\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/utils/callbacks/tensorboard.py\", line 39, in on_batch_end\n","    _log_scalars(trainer.label_loss_items(trainer.tloss, prefix='train'), trainer.epoch + 1)\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/utils/callbacks/tensorboard.py\", line 22, in _log_scalars\n","    writer.add_scalar(k, v, step)\n","  File \"/usr/local/lib/python3.10/dist-packages/torch/utils/tensorboard/writer.py\", line 391, in add_scalar\n","    self._get_file_writer().add_summary(summary, global_step, walltime)\n","  File \"/usr/local/lib/python3.10/dist-packages/torch/utils/tensorboard/writer.py\", line 113, in add_summary\n","    self.add_event(event, global_step, walltime)\n","  File \"/usr/local/lib/python3.10/dist-packages/torch/utils/tensorboard/writer.py\", line 98, in add_event\n","    self.event_writer.add_event(event)\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorboard/summary/writer/event_file_writer.py\", line 117, in add_event\n","    self._async_writer.write(event.SerializeToString())\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorboard/summary/writer/event_file_writer.py\", line 171, in write\n","    self._check_worker_status()\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorboard/summary/writer/event_file_writer.py\", line 212, in _check_worker_status\n","    raise exception\n","  File \"/usr/lib/python3.10/threading.py\", line 1016, in _bootstrap_inner\n","    self.run()\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorboard/summary/writer/event_file_writer.py\", line 244, in run\n","    self._run()\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorboard/summary/writer/event_file_writer.py\", line 275, in _run\n","    self._record_writer.write(data)\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorboard/summary/writer/record_writer.py\", line 40, in write\n","    self._writer.write(header + header_crc + data + footer_crc)\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorflow/python/lib/io/file_io.py\", line 101, in write\n","    self._writable_file.append(\n","tensorflow.python.framework.errors_impl.FailedPreconditionError: runs/detect/train15/events.out.tfevents.1690709478.098cdce72173.289.3; Transport endpoint is not connected\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 2099, in showtraceback\n","    stb = value._render_traceback_()\n","AttributeError: 'FailedPreconditionError' object has no attribute '_render_traceback_'\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1101, in get_records\n","    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 248, in wrapped\n","    return f(*args, **kwargs)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 281, in _fixed_getinnerframes\n","    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n","  File \"/usr/lib/python3.10/inspect.py\", line 1667, in getinnerframes\n","    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n","  File \"/usr/lib/python3.10/inspect.py\", line 1625, in getframeinfo\n","    filename = getsourcefile(frame) or getfile(frame)\n","  File \"/usr/lib/python3.10/inspect.py\", line 829, in getsourcefile\n","    module = getmodule(object, filename)\n","  File \"/usr/lib/python3.10/inspect.py\", line 861, in getmodule\n","    file = getabsfile(object, _filename)\n","  File \"/usr/lib/python3.10/inspect.py\", line 845, in getabsfile\n","    return os.path.normcase(os.path.abspath(_filename))\n","  File \"/usr/lib/python3.10/posixpath.py\", line 383, in abspath\n","    cwd = os.getcwd()\n","OSError: [Errno 107] Transport endpoint is not connected\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 3553, in run_code\n","    exec(code_obj, self.user_global_ns, self.user_ns)\n","  File \"\u003cipython-input-15-0c940d6c2741\u003e\", line 5, in \u003ccell line: 5\u003e\n","    results = model.train(data =data_yaml, epochs=8, imgsz=416)\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/engine/model.py\", line 377, in train\n","    self.trainer.train()\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/engine/trainer.py\", line 192, in train\n","    self._do_train(world_size)\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/engine/trainer.py\", line 354, in _do_train\n","    self.run_callbacks('on_batch_end')\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/engine/trainer.py\", line 165, in run_callbacks\n","    callback(self)\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/utils/callbacks/tensorboard.py\", line 39, in on_batch_end\n","    _log_scalars(trainer.label_loss_items(trainer.tloss, prefix='train'), trainer.epoch + 1)\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/utils/callbacks/tensorboard.py\", line 22, in _log_scalars\n","    writer.add_scalar(k, v, step)\n","  File \"/usr/local/lib/python3.10/dist-packages/torch/utils/tensorboard/writer.py\", line 391, in add_scalar\n","    self._get_file_writer().add_summary(summary, global_step, walltime)\n","  File \"/usr/local/lib/python3.10/dist-packages/torch/utils/tensorboard/writer.py\", line 113, in add_summary\n","    self.add_event(event, global_step, walltime)\n","  File \"/usr/local/lib/python3.10/dist-packages/torch/utils/tensorboard/writer.py\", line 98, in add_event\n","    self.event_writer.add_event(event)\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorboard/summary/writer/event_file_writer.py\", line 117, in add_event\n","    self._async_writer.write(event.SerializeToString())\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorboard/summary/writer/event_file_writer.py\", line 171, in write\n","    self._check_worker_status()\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorboard/summary/writer/event_file_writer.py\", line 212, in _check_worker_status\n","    raise exception\n","  File \"/usr/lib/python3.10/threading.py\", line 1016, in _bootstrap_inner\n","    self.run()\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorboard/summary/writer/event_file_writer.py\", line 244, in run\n","    self._run()\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorboard/summary/writer/event_file_writer.py\", line 275, in _run\n","    self._record_writer.write(data)\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorboard/summary/writer/record_writer.py\", line 40, in write\n","    self._writer.write(header + header_crc + data + footer_crc)\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorflow/python/lib/io/file_io.py\", line 101, in write\n","    self._writable_file.append(\n","tensorflow.python.framework.errors_impl.FailedPreconditionError: runs/detect/train15/events.out.tfevents.1690709478.098cdce72173.289.3; Transport endpoint is not connected\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 2099, in showtraceback\n","    stb = value._render_traceback_()\n","AttributeError: 'FailedPreconditionError' object has no attribute '_render_traceback_'\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 3473, in run_ast_nodes\n","    if (await self.run_code(code, result,  async_=asy)):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 3575, in run_code\n","    self.showtraceback(running_compiled_code=True)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 2101, in showtraceback\n","    stb = self.InteractiveTB.structured_traceback(etype,\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1367, in structured_traceback\n","    return FormattedTB.structured_traceback(\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1267, in structured_traceback\n","    return VerboseTB.structured_traceback(\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1124, in structured_traceback\n","    formatted_exception = self.format_exception_as_a_whole(etype, evalue, etb, number_of_lines_of_context,\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1082, in format_exception_as_a_whole\n","    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 382, in find_recursion\n","    return len(records), 0\n","TypeError: object of type 'NoneType' has no len()\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 2099, in showtraceback\n","    stb = value._render_traceback_()\n","AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1101, in get_records\n","    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 248, in wrapped\n","    return f(*args, **kwargs)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 281, in _fixed_getinnerframes\n","    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n","  File \"/usr/lib/python3.10/inspect.py\", line 1667, in getinnerframes\n","    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n","  File \"/usr/lib/python3.10/inspect.py\", line 1625, in getframeinfo\n","    filename = getsourcefile(frame) or getfile(frame)\n","  File \"/usr/lib/python3.10/inspect.py\", line 829, in getsourcefile\n","    module = getmodule(object, filename)\n","  File \"/usr/lib/python3.10/inspect.py\", line 861, in getmodule\n","    file = getabsfile(object, _filename)\n","  File \"/usr/lib/python3.10/inspect.py\", line 845, in getabsfile\n","    return os.path.normcase(os.path.abspath(_filename))\n","  File \"/usr/lib/python3.10/posixpath.py\", line 383, in abspath\n","    cwd = os.getcwd()\n","OSError: [Errno 107] Transport endpoint is not connected\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 3553, in run_code\n","    exec(code_obj, self.user_global_ns, self.user_ns)\n","  File \"\u003cipython-input-15-0c940d6c2741\u003e\", line 5, in \u003ccell line: 5\u003e\n","    results = model.train(data =data_yaml, epochs=8, imgsz=416)\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/engine/model.py\", line 377, in train\n","    self.trainer.train()\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/engine/trainer.py\", line 192, in train\n","    self._do_train(world_size)\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/engine/trainer.py\", line 354, in _do_train\n","    self.run_callbacks('on_batch_end')\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/engine/trainer.py\", line 165, in run_callbacks\n","    callback(self)\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/utils/callbacks/tensorboard.py\", line 39, in on_batch_end\n","    _log_scalars(trainer.label_loss_items(trainer.tloss, prefix='train'), trainer.epoch + 1)\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/utils/callbacks/tensorboard.py\", line 22, in _log_scalars\n","    writer.add_scalar(k, v, step)\n","  File \"/usr/local/lib/python3.10/dist-packages/torch/utils/tensorboard/writer.py\", line 391, in add_scalar\n","    self._get_file_writer().add_summary(summary, global_step, walltime)\n","  File \"/usr/local/lib/python3.10/dist-packages/torch/utils/tensorboard/writer.py\", line 113, in add_summary\n","    self.add_event(event, global_step, walltime)\n","  File \"/usr/local/lib/python3.10/dist-packages/torch/utils/tensorboard/writer.py\", line 98, in add_event\n","    self.event_writer.add_event(event)\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorboard/summary/writer/event_file_writer.py\", line 117, in add_event\n","    self._async_writer.write(event.SerializeToString())\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorboard/summary/writer/event_file_writer.py\", line 171, in write\n","    self._check_worker_status()\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorboard/summary/writer/event_file_writer.py\", line 212, in _check_worker_status\n","    raise exception\n","  File \"/usr/lib/python3.10/threading.py\", line 1016, in _bootstrap_inner\n","    self.run()\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorboard/summary/writer/event_file_writer.py\", line 244, in run\n","    self._run()\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorboard/summary/writer/event_file_writer.py\", line 275, in _run\n","    self._record_writer.write(data)\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorboard/summary/writer/record_writer.py\", line 40, in write\n","    self._writer.write(header + header_crc + data + footer_crc)\n","  File \"/usr/local/lib/python3.10/dist-packages/tensorflow/python/lib/io/file_io.py\", line 101, in write\n","    self._writable_file.append(\n","tensorflow.python.framework.errors_impl.FailedPreconditionError: runs/detect/train15/events.out.tfevents.1690709478.098cdce72173.289.3; Transport endpoint is not connected\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 2099, in showtraceback\n","    stb = value._render_traceback_()\n","AttributeError: 'FailedPreconditionError' object has no attribute '_render_traceback_'\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 3473, in run_ast_nodes\n","    if (await self.run_code(code, result,  async_=asy)):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 3575, in run_code\n","    self.showtraceback(running_compiled_code=True)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 2101, in showtraceback\n","    stb = self.InteractiveTB.structured_traceback(etype,\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1367, in structured_traceback\n","    return FormattedTB.structured_traceback(\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1267, in structured_traceback\n","    return VerboseTB.structured_traceback(\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1124, in structured_traceback\n","    formatted_exception = self.format_exception_as_a_whole(etype, evalue, etb, number_of_lines_of_context,\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1082, in format_exception_as_a_whole\n","    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 382, in find_recursion\n","    return len(records), 0\n","TypeError: object of type 'NoneType' has no len()\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 2099, in showtraceback\n","    stb = value._render_traceback_()\n","AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 3030, in _run_cell\n","    return runner(coro)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/async_helpers.py\", line 78, in _pseudo_sync_runner\n","    coro.send(None)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 3257, in run_cell_async\n","    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 3492, in run_ast_nodes\n","    self.showtraceback()\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 2101, in showtraceback\n","    stb = self.InteractiveTB.structured_traceback(etype,\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1367, in structured_traceback\n","    return FormattedTB.structured_traceback(\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1267, in structured_traceback\n","    return VerboseTB.structured_traceback(\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1142, in structured_traceback\n","    formatted_exceptions += self.format_exception_as_a_whole(etype, evalue, etb, lines_of_context,\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1082, in format_exception_as_a_whole\n","    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 382, in find_recursion\n","    return len(records), 0\n","TypeError: object of type 'NoneType' has no len()\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 2099, in showtraceback\n","    stb = value._render_traceback_()\n","AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1101, in get_records\n","    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 248, in wrapped\n","    return f(*args, **kwargs)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 281, in _fixed_getinnerframes\n","    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n","  File \"/usr/lib/python3.10/inspect.py\", line 1667, in getinnerframes\n","    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n","  File \"/usr/lib/python3.10/inspect.py\", line 1625, in getframeinfo\n","    filename = getsourcefile(frame) or getfile(frame)\n","  File \"/usr/lib/python3.10/inspect.py\", line 829, in getsourcefile\n","    module = getmodule(object, filename)\n","  File \"/usr/lib/python3.10/inspect.py\", line 861, in getmodule\n","    file = getabsfile(object, _filename)\n","  File \"/usr/lib/python3.10/inspect.py\", line 845, in getabsfile\n","    return os.path.normcase(os.path.abspath(_filename))\n","  File \"/usr/lib/python3.10/posixpath.py\", line 383, in abspath\n","    cwd = os.getcwd()\n","OSError: [Errno 107] Transport endpoint is not connected\n"]}],"source":["model = YOLO(last_weight)\n","\n","model.resume = True\n","\n","results = model.train(data =data_yaml, epochs=8, imgsz=416)"]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"background_save":true},"id":"RMw0uIvHIKIy"},"outputs":[{"name":"stderr","output_type":"stream","text":["Ultralytics YOLOv8.0.145 🚀 Python-3.10.6 torch-2.0.1+cu118 CUDA:0 (Tesla T4, 15102MiB)\n","ERROR:root:Internal Python error in the inspect module.\n","Below is the traceback from this internal error.\n","\n","ERROR:root:Internal Python error in the inspect module.\n","Below is the traceback from this internal error.\n","\n","ERROR:root:Internal Python error in the inspect module.\n","Below is the traceback from this internal error.\n","\n"]},{"name":"stdout","output_type":"stream","text":["Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 3553, in run_code\n","    exec(code_obj, self.user_global_ns, self.user_ns)\n","  File \"\u003cipython-input-16-0c940d6c2741\u003e\", line 5, in \u003ccell line: 5\u003e\n","    results = model.train(data =data_yaml, epochs=8, imgsz=416)\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/engine/model.py\", line 372, in train\n","    self.trainer = trainer(overrides=overrides, _callbacks=self.callbacks)\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/engine/trainer.py\", line 98, in __init__\n","    increment_path(Path(project) / name, exist_ok=self.args.exist_ok if RANK in (-1, 0) else True))\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/utils/files.py\", line 100, in increment_path\n","    if path.exists() and not exist_ok:\n","  File \"/usr/lib/python3.10/pathlib.py\", line 1290, in exists\n","    self.stat()\n","  File \"/usr/lib/python3.10/pathlib.py\", line 1097, in stat\n","    return self._accessor.stat(self, follow_symlinks=follow_symlinks)\n","OSError: [Errno 107] Transport endpoint is not connected: 'runs/detect/train'\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 2099, in showtraceback\n","    stb = value._render_traceback_()\n","AttributeError: 'OSError' object has no attribute '_render_traceback_'\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1101, in get_records\n","    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 248, in wrapped\n","    return f(*args, **kwargs)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 281, in _fixed_getinnerframes\n","    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n","  File \"/usr/lib/python3.10/inspect.py\", line 1667, in getinnerframes\n","    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n","  File \"/usr/lib/python3.10/inspect.py\", line 1625, in getframeinfo\n","    filename = getsourcefile(frame) or getfile(frame)\n","  File \"/usr/lib/python3.10/inspect.py\", line 829, in getsourcefile\n","    module = getmodule(object, filename)\n","  File \"/usr/lib/python3.10/inspect.py\", line 861, in getmodule\n","    file = getabsfile(object, _filename)\n","  File \"/usr/lib/python3.10/inspect.py\", line 845, in getabsfile\n","    return os.path.normcase(os.path.abspath(_filename))\n","  File \"/usr/lib/python3.10/posixpath.py\", line 383, in abspath\n","    cwd = os.getcwd()\n","OSError: [Errno 107] Transport endpoint is not connected\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 3553, in run_code\n","    exec(code_obj, self.user_global_ns, self.user_ns)\n","  File \"\u003cipython-input-16-0c940d6c2741\u003e\", line 5, in \u003ccell line: 5\u003e\n","    results = model.train(data =data_yaml, epochs=8, imgsz=416)\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/engine/model.py\", line 372, in train\n","    self.trainer = trainer(overrides=overrides, _callbacks=self.callbacks)\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/engine/trainer.py\", line 98, in __init__\n","    increment_path(Path(project) / name, exist_ok=self.args.exist_ok if RANK in (-1, 0) else True))\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/utils/files.py\", line 100, in increment_path\n","    if path.exists() and not exist_ok:\n","  File \"/usr/lib/python3.10/pathlib.py\", line 1290, in exists\n","    self.stat()\n","  File \"/usr/lib/python3.10/pathlib.py\", line 1097, in stat\n","    return self._accessor.stat(self, follow_symlinks=follow_symlinks)\n","OSError: [Errno 107] Transport endpoint is not connected: 'runs/detect/train'\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 2099, in showtraceback\n","    stb = value._render_traceback_()\n","AttributeError: 'OSError' object has no attribute '_render_traceback_'\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 3473, in run_ast_nodes\n","    if (await self.run_code(code, result,  async_=asy)):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 3575, in run_code\n","    self.showtraceback(running_compiled_code=True)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 2101, in showtraceback\n","    stb = self.InteractiveTB.structured_traceback(etype,\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1367, in structured_traceback\n","    return FormattedTB.structured_traceback(\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1267, in structured_traceback\n","    return VerboseTB.structured_traceback(\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1124, in structured_traceback\n","    formatted_exception = self.format_exception_as_a_whole(etype, evalue, etb, number_of_lines_of_context,\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1082, in format_exception_as_a_whole\n","    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 382, in find_recursion\n","    return len(records), 0\n","TypeError: object of type 'NoneType' has no len()\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 2099, in showtraceback\n","    stb = value._render_traceback_()\n","AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1101, in get_records\n","    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 248, in wrapped\n","    return f(*args, **kwargs)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 281, in _fixed_getinnerframes\n","    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n","  File \"/usr/lib/python3.10/inspect.py\", line 1667, in getinnerframes\n","    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n","  File \"/usr/lib/python3.10/inspect.py\", line 1625, in getframeinfo\n","    filename = getsourcefile(frame) or getfile(frame)\n","  File \"/usr/lib/python3.10/inspect.py\", line 829, in getsourcefile\n","    module = getmodule(object, filename)\n","  File \"/usr/lib/python3.10/inspect.py\", line 861, in getmodule\n","    file = getabsfile(object, _filename)\n","  File \"/usr/lib/python3.10/inspect.py\", line 845, in getabsfile\n","    return os.path.normcase(os.path.abspath(_filename))\n","  File \"/usr/lib/python3.10/posixpath.py\", line 383, in abspath\n","    cwd = os.getcwd()\n","OSError: [Errno 107] Transport endpoint is not connected\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 3553, in run_code\n","    exec(code_obj, self.user_global_ns, self.user_ns)\n","  File \"\u003cipython-input-16-0c940d6c2741\u003e\", line 5, in \u003ccell line: 5\u003e\n","    results = model.train(data =data_yaml, epochs=8, imgsz=416)\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/engine/model.py\", line 372, in train\n","    self.trainer = trainer(overrides=overrides, _callbacks=self.callbacks)\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/engine/trainer.py\", line 98, in __init__\n","    increment_path(Path(project) / name, exist_ok=self.args.exist_ok if RANK in (-1, 0) else True))\n","  File \"/usr/local/lib/python3.10/dist-packages/ultralytics/utils/files.py\", line 100, in increment_path\n","    if path.exists() and not exist_ok:\n","  File \"/usr/lib/python3.10/pathlib.py\", line 1290, in exists\n","    self.stat()\n","  File \"/usr/lib/python3.10/pathlib.py\", line 1097, in stat\n","    return self._accessor.stat(self, follow_symlinks=follow_symlinks)\n","OSError: [Errno 107] Transport endpoint is not connected: 'runs/detect/train'\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 2099, in showtraceback\n","    stb = value._render_traceback_()\n","AttributeError: 'OSError' object has no attribute '_render_traceback_'\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 3473, in run_ast_nodes\n","    if (await self.run_code(code, result,  async_=asy)):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 3575, in run_code\n","    self.showtraceback(running_compiled_code=True)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 2101, in showtraceback\n","    stb = self.InteractiveTB.structured_traceback(etype,\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1367, in structured_traceback\n","    return FormattedTB.structured_traceback(\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1267, in structured_traceback\n","    return VerboseTB.structured_traceback(\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1124, in structured_traceback\n","    formatted_exception = self.format_exception_as_a_whole(etype, evalue, etb, number_of_lines_of_context,\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1082, in format_exception_as_a_whole\n","    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 382, in find_recursion\n","    return len(records), 0\n","TypeError: object of type 'NoneType' has no len()\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 2099, in showtraceback\n","    stb = value._render_traceback_()\n","AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 3030, in _run_cell\n","    return runner(coro)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/async_helpers.py\", line 78, in _pseudo_sync_runner\n","    coro.send(None)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 3257, in run_cell_async\n","    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 3492, in run_ast_nodes\n","    self.showtraceback()\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 2101, in showtraceback\n","    stb = self.InteractiveTB.structured_traceback(etype,\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1367, in structured_traceback\n","    return FormattedTB.structured_traceback(\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1267, in structured_traceback\n","    return VerboseTB.structured_traceback(\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1142, in structured_traceback\n","    formatted_exceptions += self.format_exception_as_a_whole(etype, evalue, etb, lines_of_context,\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1082, in format_exception_as_a_whole\n","    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 382, in find_recursion\n","    return len(records), 0\n","TypeError: object of type 'NoneType' has no len()\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\", line 2099, in showtraceback\n","    stb = value._render_traceback_()\n","AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 1101, in get_records\n","    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 248, in wrapped\n","    return f(*args, **kwargs)\n","  File \"/usr/local/lib/python3.10/dist-packages/IPython/core/ultratb.py\", line 281, in _fixed_getinnerframes\n","    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n","  File \"/usr/lib/python3.10/inspect.py\", line 1667, in getinnerframes\n","    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n","  File \"/usr/lib/python3.10/inspect.py\", line 1625, in getframeinfo\n","    filename = getsourcefile(frame) or getfile(frame)\n","  File \"/usr/lib/python3.10/inspect.py\", line 829, in getsourcefile\n","    module = getmodule(object, filename)\n","  File \"/usr/lib/python3.10/inspect.py\", line 861, in getmodule\n","    file = getabsfile(object, _filename)\n","  File \"/usr/lib/python3.10/inspect.py\", line 845, in getabsfile\n","    return os.path.normcase(os.path.abspath(_filename))\n","  File \"/usr/lib/python3.10/posixpath.py\", line 383, in abspath\n","    cwd = os.getcwd()\n","OSError: [Errno 107] Transport endpoint is not connected\n"]}],"source":[]}],"metadata":{"accelerator":"GPU","colab":{"authorship_tag":"ABX9TyPP9dCOI0jWDxGt9CH9vvBH","machine_shape":"hm","name":"","version":""},"kernelspec":{"display_name":"Python 3","name":"python3"},"language_info":{"name":"python"}},"nbformat":4,"nbformat_minor":0}